### PR TITLE
improve iot config example

### PR DESCRIPTION
**Pull request by jbtrystram**
### Type of change

- Enhancement / new feature

### Description

detail the adapters config in iot config, so one can easily disable them if necessary.

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: templates/iot/examples/iot-config.yaml [hunk 1]
```diff
@@ -19,3 +19,18 @@ spec:
       endpoint:
         secretNameStrategy:
           secretName: iot-mqtt-adapter-tls
+    http:
+      enabled: true
+      replicas: 3
```
Comment (dejanb) [inline templates/iot/examples/iot-config.yaml]: I would keep default values here. Just make them present to be easy to change, or move to other adapters

File: templates/iot/examples/iot-config.yaml [hunk 2]
```diff
@@ -19,3 +19,18 @@ spec:
       endpoint:
         secretNameStrategy:
           secretName: iot-mqtt-adapter-tls
```
Comment (dejanb) [inline templates/iot/examples/iot-config.yaml]: enabled for mqtt?

**Comment by jbtrystram** [comment]
@dejanb do you think I should add `tenantIdleTimeout` and `maxPayloadSize` ?

**Comment by dejanb** [comment]
Yeah, let's add them and add the documentation similar to https://enmasse.io/documentation/master/openshift/#address-space-examples-messaging

**Comment by jbtrystram** [comment]
@dejanb WDYT ?

File: documentation/modules/ref-iot-config-example-adapters.adoc [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Module included in the following assemblies:
+//
+// assembly-iot-service-admin-guide.adoc
+
+[id='iot-adapters-example-{context}']
+= Iot adapters example
+
+This iot config example shows how the protocol adapters can be individually tuned.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-config-example-adapters.adoc]:
```suggestion
This IoT configuration example shows how the protocol adapters can be individually tuned.
```

File: documentation/modules/ref-iot-config-example-adapters.adoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// Module included in the following assemblies:
+//
+// assembly-iot-service-admin-guide.adoc
+
+[id='iot-adapters-example-{context}']
+= Iot adapters example
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-config-example-adapters.adoc]:
```suggestion
= IoT adapters example
```

File: documentation/modules/ref-iot-config-example-adapters.adoc [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+This iot config example shows how the protocol adapters can be individually tuned.
+
+[source,yaml,options="nowrap"]
+----
+include::../common/iot-config-example1.yaml[]
+----
+<1> Protocols adapters can be disabled if necessary. Default value is `true`.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-config-example-adapters.adoc]:
```suggestion
<1> Protocol adapters can be disabled if necessary. The default value is `true`.
```

File: documentation/modules/ref-iot-config-example-adapters.adoc [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+<2> The duration for which client connection are kept alive for.
+<3> The maximum allowed size of an incoming message in bytes.
+<4> Containers resources and instances can be adjusted if necessary.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-config-example-adapters.adoc]:
```suggestion
<4> Container resources and instances can be adjusted if necessary.
```

File: documentation/modules/ref-iot-config-example.adoc [hunk 7]
```diff
@@ -0,0 +1,16 @@
+// Module included in the following assemblies:
+//
+// assembly-iot-service-admin-guide.adoc
+
+[id='iotconfig-example-{context}']
+= Iot Config example
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-config-example.adoc]:
```suggestion
= IoT configuration example
```

File: documentation/modules/ref-iot-config-example.adoc [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+This iot config example shows only the required options to create an `iotConfig`.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-config-example.adoc]:
```suggestion
This IoT configuration example shows only the required options to create an `iotConfig`.
```

File: documentation/modules/ref-iot-config-example.adoc [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+[source,yaml,options="nowrap"]
+----
+include::../common/iot-config-default.yaml[]
+----
+<1> The infinispan service must be provided.
```
Comment (jenmalloy) [inline documentation/modules/ref-iot-config-example.adoc]:
```suggestion
<1> The Infinispan service must be provided.
```

Comment (ctron) [reply documentation/modules/ref-iot-config-example.adoc]: I think we might need a variable here: "Infinispan" becomes "Red Hat Data Grid" in downstream.

Comment (jbtrystram) [reply documentation/modules/ref-iot-config-example.adoc]: @ctron good catch ! Updated

File: documentation/assemblies/assembly-iot-service-admin-guide.adoc [hunk 10]
```diff
@@ -11,4 +11,6 @@ Service administrators typically install and configure the Internet of Things (I
 
 include::../modules/proc-iot-installing-services.adoc[leveloffset=+1]
 
+include::../modules/ref-iot-config-example.adoc[leveloffset=+1]
 
+include::../modules/ref-iot-config-example-adapters.adoc[leveloffset=+1]
```
Comment (dejanb) [inline documentation/assemblies/assembly-iot-service-admin-guide.adoc]: I would put all these under one section "IoT configuration examples". And then have two subsections: "Minimal IoT configuration example" and "IoT configuration example for tuning protocol adapters"

Comment (jenmalloy) [reply documentation/assemblies/assembly-iot-service-admin-guide.adoc]: @dejanb, good point, but I'd keep the files separate and create a new assembly called "Examples" that included both files.

**Comment by dejanb** [comment]
This is now included in #3606 with other IoT docs changes

**Pull request closed without merging**